sycl/include/sycl/detail/info_desc_helpers.hpp

```diff
@@ -81,20 +81,7 @@ struct IsSubGroupInfo<info::kernel_device_specific::compile_sub_group_size>
   struct is_##DescType##_info_desc<info::DescType::Desc> : std::true_type {    \
     using return_type = info::DescType::Desc::return_type;                     \
   };
-#define __SYCL_PARAM_TRAITS_SPEC_WITH_INPUT(DescType, Desc, ReturnT, InputT,   \
-                                            PiCode)                            \
-  template <> struct PiInfoCode<info::DescType::Desc> {                        \
-    static constexpr                                                           \
-        typename std::conditional<IsSubGroupInfo<info::DescType::Desc>::value, \
-                                  pi_kernel_sub_group_info,                    \
-                                  pi_kernel_group_info>::type value = PiCode;  \
-  };                                                                           \
-  template <>                                                                  \
-  struct is_##DescType##_info_desc<info::DescType::Desc> : std::true_type {    \
-    using with_input_return_type = info::DescType::Desc::return_type;          \
-  };
```

Comment:
Having a removal in the PR that adds new functionality and deprecates an old one means we're doing two things in one PR - refactoring *and* the functional change in question. I'd suggest to create two PRs next time.

Or am I missing anything?

Reply:
The deprecation would end up just being 1 line, since the other changes are needed to make the actual SYCL 2020 version behave.